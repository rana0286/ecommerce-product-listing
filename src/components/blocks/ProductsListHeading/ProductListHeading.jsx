import React from 'react';
import PropTypes from 'prop-types';
import { Heading } from '../../elements';

const ProductListHeading = ({ headingLabel }) => <Heading variation="h2">{headingLabel}</Heading>;

ProductListHeading.propTypes = {
  headingLabel: PropTypes.string.isRequired,
};

export default ProductListHeading;
