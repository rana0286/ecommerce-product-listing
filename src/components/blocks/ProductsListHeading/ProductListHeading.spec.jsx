import React from 'react';
import { mount } from 'enzyme';
import ProductListHeading from './ProductListHeading';
import renderer from 'react-test-renderer';

describe('src/components/blocks/ProductListHeading', () => {
  it('ProductListHeading component should render with h2', () => {
    const component = mount(<ProductListHeading headingLabel="Test" />);
    const tree = renderer.
      create(<ProductListHeading headingLabel="Test" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(component.find('h2').length).toBe(1);
  });
});
