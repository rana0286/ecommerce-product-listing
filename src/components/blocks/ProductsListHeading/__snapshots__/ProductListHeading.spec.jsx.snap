// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/blocks/ProductListHeading ProductListHeading component should render with h2 1`] = `
<h2
  className="heading heading--h2"
>
  Test
</h2>
`;
